src/Umbraco.Core/Services/Implement/DomainService.cs

```diff
@@ -91,11 +92,55 @@ public Attempt<OperationResult> Save(IDomain domainEntity)
                 if (scope.Events.DispatchCancelable(Saving, this, saveEventArgs))
                 {
                     scope.Complete();
+
                     return OperationResult.Attempt.Cancel(evtMsgs);
                 }
 
                 _domainRepository.Save(domainEntity);
                 scope.Complete();
+
+                saveEventArgs.CanCancel = false;
+                scope.Events.Dispatch(Saved, this, saveEventArgs);
+            }
+
+            return OperationResult.Attempt.Succeed(evtMsgs);
+        }
+
+        public Attempt<OperationResult> Sort(IEnumerable<IDomain> items)
```

Comment:
This works exactly like media: if the sort order doesn't match the index of the items, it's updated and saved.